File: java/src/test/java/no/bouvet/workshop/solid/solution/ConsumerCsvReader.java
package no.bouvet.workshop.solid.solution;

import no.bouvet.workshop.solid.a1_output.CsvConsumer;
import org.junit.Test;

import java.io.File;

public class ConsumerCsvReader extends OpenClosedCsvReader {

    private final CsvConsumer csvConsumer;

    public ConsumerCsvReader(CsvConsumer csvConsumer) {
        this.csvConsumer = csvConsumer;
    }

    @Override
    protected void visitLine(String formatted) {
        csvConsumer.visitLine(formatted);
    }

    @Override
    protected void visitEnd(File file) {
        csvConsumer.visitEnd(file);
    }

    @Override
    protected void visitStart(File file) {
        csvConsumer.visitStart(file);
    }
}

File: dotNet/delegation/test/Solution/ConsumerCsvReader.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using test.a1_output;

namespace test.Solution
{
    public class ConsumerCsvReader : OpenClosedCsvReader
    {
        private readonly CsvConsumer csvConsumer;

        public ConsumerCsvReader(CsvConsumer csvConsumer)
        {
            this.csvConsumer = csvConsumer;
        }

        protected override void visitLine(string formatted)
        {
            csvConsumer.visitLine(formatted);
        }

        protected override void visitEnd(FileInfo file)
        {
            csvConsumer.visitEnd(file);
        }

        protected override void visitStart(FileInfo file)
        {
            csvConsumer.visitStart(file);
        }
    }
}
